logstreams/src/main/java/io/zeebe/distributedlog/restore/impl/DefaultRestoreInfoRequestHandler.java

```diff
@@ -37,23 +38,31 @@ public DefaultRestoreInfoRequestHandler(
   }
 
   @Override
-  public RestoreInfoResponse onRestoreInfoRequest(RestoreInfoRequest request) {
-    final ReplicationTarget target;
+  public RestoreInfoResponse onRestoreInfoRequest(RestoreInfoRequest request, Logger logger) {
+    RestoreInfoResponse response = DefaultRestoreInfoResponse.NONE;
     final long lastValidSnapshotPosition = snapshotController.getLastValidSnapshotPosition();
+
+    logger.debug("Received restore info request {}", request);
     if (lastValidSnapshotPosition > -1
         && lastValidSnapshotPosition >= request.getLatestLocalPosition()) {
-      target = RestoreInfoResponse.ReplicationTarget.SNAPSHOT;
       final File lastValidSnapshotDirectory = snapshotController.getLastValidSnapshotDirectory();
-      final int numChunks = lastValidSnapshotDirectory.listFiles().length;
-      return new DefaultRestoreInfoResponse(
-          target, new DefaultSnapshotRestoreInfo(lastValidSnapshotPosition, numChunks));
+      final File[] chunks = lastValidSnapshotDirectory.listFiles();
+
+      if (chunks != null && chunks.length > 0) {
+        response =
+            new DefaultRestoreInfoResponse(
+                ReplicationTarget.SNAPSHOT,
+                new DefaultSnapshotRestoreInfo(lastValidSnapshotPosition, chunks.length));
+      }
     } else if (seekToRequestedPositionExclusive(request.getLatestLocalPosition())) {
-      target = RestoreInfoResponse.ReplicationTarget.EVENTS;
-    } else {
-      target = RestoreInfoResponse.ReplicationTarget.NONE;
+      response = new DefaultRestoreInfoResponse(ReplicationTarget.EVENTS);
     }
 
-    return new DefaultRestoreInfoResponse(target);
+    logger.debug(
```

Comment:
I think it will be useful to log the position of last event in the logstream, if the response is NONE.

Reply:
Hm, this is helpful if we're trying to replicate an event that hasn't been written yet, but not so much if it's a missing event (i.e. lower than latest event). But it would be useful in that case, I guess.